File: Python/Problem 7.py
from math import sqrt
def ok(x):
    stp = int(sqrt(x))+1
    for i in range(2,stp,1):
        if ( x%i==0 ):
            return 0
    return 1
i = 1
cnt = 0
while ( cnt<10001 ):
    i+=1
    if ( ok(i) ):
        cnt+=1
print i

File: Cplusplus/Problem 7.cpp
# include <cstdio>
# include <iostream>
using namespace std;
int cnt = 0;
int ok ( int x )
{
    for ( int i = 2;i*i <=x;i++ ) {
        if ( x%i==0 ) return 0;
    }
    cnt++;
    return 1;
}
int main(void)
{
    for ( int i = 2;;i++ ) {
        if ( ok(i)&&cnt==10001 ) {
            cout<<i<<endl;
            break;
        }
    }
    return 0;
}
